x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/aggregations/support/HistogramValuesSourceType.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.analytics.aggregations.support;
+
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.IndexHistogramFieldData;
+import org.elasticsearch.script.AggregationScript;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+import org.elasticsearch.search.aggregations.support.CoreValuesSourceType;
+import org.elasticsearch.search.aggregations.support.FieldContext;
+import org.elasticsearch.search.aggregations.support.ValueType;
+import org.elasticsearch.search.aggregations.support.ValuesSource;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+
+import java.util.Locale;
+import java.util.function.LongSupplier;
+
+public enum HistogramValuesSourceType implements ValuesSourceType {
```

Comment:
+1 for using an enum here.  I'm actually about to update the documentation on `ValuesSourceType` to recommend this pattern.  That said, I'd suggest `AnalyticsValuesSourceType` as the name, in case we want to add another here later.

Reply:
/cc @csoulios who I chatted with briefly about this.  I guess it comes down to how we want to organize new VSTypes?  E.g. are they grouped by plugin/module (everything in Analytics goes into one enum), or based on functionality (e.g. Rollup has it's own enum, Histo has it's own, etc)?

Probably doesn't really matter other than being consistent.